docs/dev/Pagination-v2.md

```diff
@@ -114,67 +496,75 @@ sequenceDiagram
+1. `Deserialization` is performed by `PlanSerializer` to restore entire Physical Plan Tree encoded into the cursor.
+2. Since query already contains the Physical Plan Tree, all tree processing steps are skipped.
+3. `Serialization` is performed by `PlanSerializer` - it converts Physical Plan Tree into a cursor, which could be used query a next page.
+4. Traversal of Physical Plan Tree to get total hits, which is required to properly fill response to a user.
+
 ```mermaid
 sequenceDiagram
     participant SQLService
     participant QueryPlanFactory
     participant QueryService
     participant OpenSearchExecutionEngine
     participant PlanSerializer
-    participant Physical Plan Tree
 
-SQLService->>+QueryPlanFactory: execute
-  QueryPlanFactory->>+QueryService: execute
-    critical Deserialization
-    QueryService->>+PlanSerializer: convertToPlan
-      PlanSerializer-->>-QueryService: Physical plan tree
+SQLService ->>+ QueryPlanFactory : execute
+  QueryPlanFactory ->>+ QueryService : execute
+    rect rgb(191, 223, 255)
+    note over QueryService, PlanSerializer : Deserialization
+    QueryService ->>+ PlanSerializer: convertToPlan
+      PlanSerializer -->>- QueryService: Physical plan tree
     end
-    Note over QueryService: Planner, Optimizer and Implementor<br />are skipped
-    QueryService->>+OpenSearchExecutionEngine: execute
-      Note over OpenSearchExecutionEngine: iterate result set
-      critical Serialization
-      OpenSearchExecutionEngine->>+PlanSerializer: convertToCursor
-        PlanSerializer-->>-OpenSearchExecutionEngine: cursor
+    Note over QueryService : Planner, Optimizer and Implementor<br />are skipped
+    QueryService ->>+ OpenSearchExecutionEngine : execute
+      rect rgb(191, 223, 255)
+      note over OpenSearchExecutionEngine, PlanSerializer : Serialization
+      OpenSearchExecutionEngine ->>+ PlanSerializer : convertToCursor
+        PlanSerializer -->>- OpenSearchExecutionEngine : cursor
       end
-      critical
-      OpenSearchExecutionEngine->>+Physical Plan Tree: getTotalHits
-        Physical Plan Tree-->>-OpenSearchExecutionEngine: total hits
+      rect rgb(191, 223, 255)
+      Note over OpenSearchExecutionEngine : get total hits
       end
-      OpenSearchExecutionEngine-->>-QueryService: execution completed
-    QueryService-->>-QueryPlanFactory: execution completed
-  QueryPlanFactory-->>-SQLService: execution completed
+      OpenSearchExecutionEngine -->>- QueryService: execution completed
+    QueryService -->>- QueryPlanFactory : execution completed
+  QueryPlanFactory -->>- SQLService : execution completed
 ```
-### Legacy Engine Fallback
+
+#### Legacy Engine Fallback
+
+Since pagination in V2 engine supports less SQL features than pagination in legacy engine, a fallback mechanism is created to keep V1 engine features still available for the end user. Pagination fallback is backed by a new exception type which allows legacy engine to intersect execution of a request.
```

Comment:
```suggestion
Since pagination in V2 engine supports fewer SQL commands than pagination in legacy engine, a fallback mechanism is created to keep V1 engine features still available for the end user. Pagination fallback is backed by a new exception type which allows legacy engine to intersect execution of a request.
```